silnlp/nmt/translate.py

```diff
@@ -35,6 +62,52 @@ def _insert_lang_tag(self, text: str, trg_iso: Optional[str]) -> str:
             return f"<2{trg_iso}> {text}"
         return text
 
+    def translate_book_by_step(self, book, src_project=None, output_usfm=None, trg_lang=None):
+        self.init_translation_task(experiment_suffix=f"_{self.checkpoint}_{book}")
+        if src_project is None:
+            if len(self.config.src_projects) != 1:
+                raise RuntimeError("A source project must be specified.")
+            src_project = next(iter(self.config.src_projects))
+
+        default_output_dir = self.config.exp_dir / "infer" / self._step_str
+        output_path: Optional[Path] = None if output_usfm is None else Path(output_usfm)
+        if output_path is None:
+            book_num = book_id_to_number(book)
+            output_path = default_output_dir / f"{book_file_name_digits(book_num)}{book}.SFM"
+        elif output_path.name == output_path:
+            output_path = default_output_dir / output_path
+
+        output_path.parent.mkdir(exist_ok=True, parents=True)
+        self.translate_book(src_project, book, output_path, trg_iso=trg_lang)
+        SIL_NLP_ENV.copy_experiment_to_bucket(self.name)
+
+    def translate_parts(self, src_prefix, trg_prefix, start_seq, end_seq, trg_lang=None):
+        self.init_translation_task(experiment_suffix=f"_{self.checkpoint}_{src_prefix}")
+        if trg_prefix is None:
+            raise RuntimeError("A target file prefix must be specified.")
+        if start_seq is None or end_seq is None:
+            raise RuntimeError("Start and end sequence numbers must be specified.")
+        cwd = Path.cwd()
+        for i in range(start_seq, end_seq + 1):
+            file_num = f"{i:04d}"
+            src_file_path = cwd / f"{src_prefix}{file_num}.txt"
+            trg_file_path = cwd / f"{trg_prefix}{file_num}.txt"
+            if src_file_path.is_file() and not trg_file_path.is_file():
+                start = time.time()
+                self.translate_text_file(src_file_path, trg_file_path, trg_iso=trg_lang)
+                end = time.time()
+                print(f"Translated {src_file_path.name} to {trg_file_path.name} in {((end-start)/60):.2f} minutes")
+        SIL_NLP_ENV.copy_experiment_to_bucket(self.name)
+
+    def translate_src(self, src_file_path, trg_lang=None, trg_file_path=None):
```

Comment:
I would name this `translate_text_file`.

Reply:
Done.